### PR TITLE
feat: implement RequestTimeout in BackendTrafficPolicy

### LINKED ISSUE #3251: Setting to override default request timeout
*Description*:
The default timeout looks to be 10seconds we would like to easily be able to change this globally to 30 seconds. It maybe possible to do this with envoy config patching but this seems overly complicated for a simple change.

### PR DESCRIPTION
**Pull request by sanposhiho**
**What type of PR is this?**

**What this PR does / why we need it**:

implement RequestTimeout in BackendTrafficPolicy

**Which issue(s) this PR fixes**:

Fixes #3251

### CODE REVIEW

File: internal/gatewayapi/clustersettings.go [hunk 1]
```diff
@@ -128,9 +129,19 @@ func buildClusterSettingsTimeout(policy egv1a1.ClusterSettings, traffic *ir.Traf
...
+				errs = errors.Join(errs, fmt.Errorf("invalid RequestTimeout value %s", *pto.HTTP.RequestTimeout))
+			} else {
+				rt = ptr.To(metav1.Duration{Duration: d})
+			}
+		}
+
 		to.HTTP = &ir.HTTPTimeout{
 			ConnectionIdleTimeout: cit,
 			MaxConnectionDuration: mcd,
+			RequestTimeout:        rt,
```
Comment (arkodg) [inline internal/gatewayapi/clustersettings.go:144]: not seeing any changes in https://github.com/envoyproxy/gateway/blob/77c042cc146013bc731a27cbc4e7970610c31535/internal/gatewayapi/clustersettings.go#L149
can we ensure HTTPRoute Request Timeout takes precedence over BTP HTTP Request Timeout using a yaml test case?

**Review by arkodg — DISMISSED** [review]
LGTM thanks !

**Comment by arkodg** [comment]
hey can you run `make generate` and commit those changes ?

**Comment by sanposhiho** [comment]
oops, done 🙏

**Comment by arkodg** [comment]
can you re run `make generate` and commit those changes @sanposhiho , looks like you will not be able to add a custom comment like `# Overwritten by the request timeout in HTTPRoute` since it breaks the test generation check

**Comment by sanposhiho** [comment]
@arkodg Now it's fixed! (looks like I needed rebase + `make manifests`)

**Review by arkodg — APPROVED** [review]

**Review by guydc — APPROVED** [review]
LGTM

**Pull request merged**